tests/test_annotations.py

```diff
@@ -0,0 +1,337 @@
+        data["http://www.w3.org/ns/oa#motivatedBy"] = [{
+            "@id": Annotation.IDLING
+        }]
+        data["http://www.w3.org/ns/oa#hasBody"] = [{
+            "@type" : ["http://www.w3.org/ns/oa#TextualBody"],
+            "http://www.w3.org/ns/oa#bodyValue": [{
+                "@value": "A good description of the topic that bears further investigation"
+            }],
+            "http://www.w3.org/ns/oa#hasPurpose": [{
+                "@id": "http://www.w3.org/ns/oa#describing"
+            }]
+        }]
+        data["http://www.w3.org/ns/oa#hasTarget"] = [{
+            "http://www.w3.org/ns/oa#hasSource": [{
+                "@id": self.identifier.urn
+            }],
+            "http://www.w3.org/ns/oa#hasSelector": [{
+                "@type": ["http://www.w3.org/ns/oa#FragmentSelector"],
+                "http://www.w3.org/1999/02/22-rdf-syntax-ns#value": [{
+                    "@value": "epubcfi(/6/4[chap01ref]!/4[body01]/10[para05]/3:10)"
+                }]
+            }]
+        }]
+
+        data = json.dumps(data)
+
+        annotation = AnnotationParser.parse(self._db, data, self.default_patron)
+        eq_(self.default_patron.id, annotation.patron_id)
+        eq_(self.identifier.id, annotation.identifier_id)
+        eq_(Annotation.IDLING, annotation.motivation)
+        eq_(True, annotation.active)
+
+    def test_parse_compacted_jsonld(self):
+        self.pool.loan_to(self.default_patron)
+
+        data = dict()
+        data["@type"] = "http://www.w3.org/ns/oa#Annotation"
+        data["http://www.w3.org/ns/oa#motivatedBy"] = {
+            "@id": Annotation.IDLING
+        }
+        data["http://www.w3.org/ns/oa#hasBody"] = {
+            "@type": "http://www.w3.org/ns/oa#TextualBody",
+            "http://www.w3.org/ns/oa#bodyValue": "A good description of the topic that bears further investigation",
+            "http://www.w3.org/ns/oa#hasPurpose": {
+                "@id": "http://www.w3.org/ns/oa#describing"
+            }
+        }
+        data["http://www.w3.org/ns/oa#hasTarget"] = {
+            "http://www.w3.org/ns/oa#hasSource": {
+                "@id": self.identifier.urn
+            },
+            "http://www.w3.org/ns/oa#hasSelector": {
+                "@type": "http://www.w3.org/ns/oa#FragmentSelector",
+                "http://www.w3.org/1999/02/22-rdf-syntax-ns#value": "epubcfi(/6/4[chap01ref]!/4[body01]/10[para05]/3:10)"
+            }
+        }
+
+        data = json.dumps(data)
+
+        annotation = AnnotationParser.parse(self._db, data, self.default_patron)
+        eq_(self.default_patron.id, annotation.patron_id)
+        eq_(self.identifier.id, annotation.identifier_id)
+        eq_(Annotation.IDLING, annotation.motivation)
+        eq_(True, annotation.active)
+
+    def test_parse_jsonld_with_context(self):
+        self.pool.loan_to(self.default_patron)
+
+        data = dict()
+        data["@context"] = [AnnotationWriter.JSONLD_CONTEXT, 
```

Comment:
Does it make sense to refactor this into a helper method? You generate some very similar documents here.